modules/contact/include/userobjects/WeightedGapUserObject.h

```diff
@@ -109,6 +109,12 @@ class WeightedGapUserObject : public MortarUserObject
   const ADReal * _weighted_gap_ptr = nullptr;
   const Real * _normalization_ptr = nullptr;
 
+  /// Whether to use Petrov-Galerkin approach
+  const bool _use_petrov_galerkin;
+
+  /// The auxiliary Lagrange multiplier variable (used together whith the Petrov-Galerkin approach)
+  MooseVariable * const _aux_lm_var;
```

Comment:
This shouldn't go in the base class as we may be using the penalty approach which does not have a Lagrange multiplier

Reply:
Why was this marked as resolved? I'm guessing there's a problem due to the fact that `LMWeightedVelocitiesUserObject` doesn't inherit from `LMWeightedGapUserObject`?

Reply:
Oh I guess I thought this to be resolved when I first addressed you guys comments yesterday, after which we had the discussion about where to put it and I forgot about it. Thanks for your reminder.

The `LMWeightedVelocitiesUserObject ` is inherited from `WeightedVelocitiesUserObject` which is inherited from `WeightedGapUserObject`. And `LMWeightedGapUserObject ` is directly inherited from `WeightedGapUserObject`.

So I am not sure what to change here. Do you suggest adding `_use_petrov_galerkin` and `_aux_lm_var` members and associated options twice in the two children classes?

Reply:
And obviously I should have made the aux variable const..

Reply:
Okay I get it after bringing this up with @recuero during our meeting. Basically no PG option (nor aux_lm_var) should appear in classes not starting with LM. Let me make the change

Reply:
I just mostly wanted to have more discussion about it. I agree with the conclusion you reached with @recuero. And then this duplication will be removed in the refactoring @dschwen promised 😄